cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -106,10 +106,11 @@ public static JmxTrans fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions
                 throw new InvalidResourceException(error);
             }
             result = new JmxTrans(kafkaAssembly.getMetadata().getNamespace(),
-                    kafkaAssembly.getMetadata().getName(),
-                    Labels.fromResource(kafkaAssembly).withKind(kafkaAssembly.getKind()));
+                    kafkaAssembly.getMetadata().getName());
             result.isDeployed = true;
 
+            result.setDefaultLabels(kafkaAssembly);
```

Comment:
Is there some object inheriting from `AbtractModel` which does not do this? If yes, why does it need to be separate method and not part of the constructor?

Reply:
So I had a long time thinking about this. Originally I didn't include it as I did not want to add a `HasMetadata` as an argument, but also it sort of made sense to me that since all other label set fields are defined outside of the constructor this one should be also...

Happy to move it in if you disagree with this approach

Reply:
I can probably live with this if you think it is better. I there is a risk that one day someone will forget to add it to some new model. However, as for adding the `HasMetadata`to the constructor ... it seesm in most cases you call it anyway with something like this as parameters: `kafkaAssembly.getMetadata().getNamespace(), kafkaAssembly.getMetadata().getName()`. So maybe the constructor could have only `HasMetadata` and get the name, namespace and labels out of it on its own.

Reply:
Another option would be to define an `abstract` getter in `AbstractModel`, forcing subclasses to override it. I actually think this is preferable, since there's no need for this property to be mutable AFAICT.